### ISSUE TITLE
using 0.12.7_patch2 followed by 0.12.8 results in "database updated by newer version" error

### ISSUE THREAD

**Issue by mmhere** [AsamK/signal-cli#1442]
> Note, upon using the patch, then using official 0.12.8, this error occurs:
> 
> ```
> ERROR AccountDatabase - Database has been updated by a newer signal-cli version
> Error loading state file for user +15032240052: java.sql.SQLException: Database has been updated by a newer signal-cli version (RuntimeException)
> ```
> 
> Apparently the internal versioning doesn't quite agree, with the patch posted earlier representing itself as a "newer" version than the official release.
> 
> So if you went from 0.12.7 to 0.12.8 I imagine you'd be OK, but if you used 0.12.8_patch2 after 0.12.7 then tried to go to 0.12.8, you will have problems. At least I did.
> 
> I thus had to go back to the 0.12.7_patch2 version and will continue using it until some future time.

I cannot use 0.12.8 without re-registering, it seems.  any workaround for this?

I think maybe this issue should stay open for a bit.

_Originally posted by @mmhere in https://github.com/AsamK/signal-cli/issues/1439#issuecomment-1928520727_

**Comment by ced-b** [comment]
What happened is that I tagged `0.12.7_patch2` at the HEAD of the repository. This version of the code already contained more changes than just the immediate fix to the immediate problem. It seems to have included changes to the database files that are not backwards compatible. 

When @AsamK made the tag for `0.12.8` he made a small fix to the `0.12.7` code but did not yet include any other changes that are at the HEAD of the repository. So in a way `0.12.8` is behind `0.12.7_patch2`.

Unless you have a specific problem it may be simplest to wait until the `0.12.9` release. That's what I am doing.

**Comment by mmhere** [comment]
Yeah, I figured I'd just use the patch and await the next official release.  I assumed the versioning was exactly as @ced-b described.  Thanks!

**Comment by AsamK** [comment]
Yeah, I just made a small 0.12.8 release to work around the server side version check. The next major release will be 0.13.0 which will be released soonish.

**Comment by sparrowme** [comment]
The database version I am on is 23 per -vv when I had a compiled the 12.8_SNAPSHOT... but I attempted to fix another issue by recompiling the latest native but now that compile fails.  

I tried to move to the pre-compiled native and the mismatch is similar as reported in OP.

How do I get it back running or do I have to wipe my accounts on the cli and start over?
```
[main] TRACE o.a.s.m.storage.AccountDatabase - Current database version: 23 Program database version: 18
```

**Comment by m-ueberall** [comment]
> The database version I am on is 23 per -vv when I had a compiled the 0.12.8_SNAPSHOT... […]
> I tried to move to the pre-compiled native and the mismatch is similar as reported in OP.
> How do I get it back running or do I have to wipe my accounts on the cli and start over?

By "moving to the pre-compiled native", I assume you mean a pre-compiled binary based on a newer codebase than 0.12.8*SNAPSHOT from this repository, that is, 0.12.9-SNAPSHOT ([see the attached artefacts on the bottom of this page](https://github.com/AsamK/signal-cli/actions/runs/7854468767))?
If _yes,_ then you're likely out of luck.
If _not,_ then you probably referred to the newly released 0.12.8 (w/o suffix) which is older than 0.12.7_patch2 from the third-party repository/temporary fork as the latter stems from the same code branch as 0.12.9-SNAPSHOT above.

**Comment by sparrowme** [comment]
I was successful in compiling to a specific commit fb85e1a to get back up and running. thaniks

latest commit as of now still fails to compile for me :shrug:
